tests/tpem_encoder

```diff
@@ -0,0 +1,161 @@
+#!/bin/bash -e
+# Copyright (C) 2022 Simo Sorce <simo@redhat.com>
+# SPDX-License-Identifier: Apache-2.0
+
+source "${TESTSSRCDIR}/helpers.sh"
+
+# We need to configure early loading otherwise no digests are loaded,
+# and all checks are skipped
+sed -e "s/#pkcs11-module-encode-provider-uri-to-pem/pkcs11-module-encode-provider-uri-to-pem = true/" \
+    -e "s/#pkcs11-module-load-behavior/pkcs11-module-load-behavior = early/" \
+    "${OPENSSL_CONF}" > "${OPENSSL_CONF}.encode_to_pem"
+OPENSSL_CONF=${OPENSSL_CONF}.encode_to_pem
+make-uri-pem() {
+    export LC_ALL=C
+
+    URI=$1
+    OUT=$2
+    DESC="${3:-PKCS#11 Provider URI v1.0}"
+
+    DESC_HEX=$(printf '%s' "${DESC}" | perl -ne 'print unpack "H*", $_')
```

Comment:
Perl is a build dependency of openssl, so I thought it's ok to use. I can't remember why, but I had an issue with xxd (I think it behaves somehow differently on macos). I'm going to change it such that it only uses printf and od.

Reply:
I have a personal dislike of perl :-D

Reply:
od and printf are used instead of perl pack/unpack.
```
DESC_HEX=$(printf '%s' "${DESC}" | od -An -t x1)
...
# shellcheck disable=SC2059 # printf should treat variable as format string
printf "${PEM_HEX}" | base64
```